worker/uniter/runner/jujuc/secret-add.go

```diff
@@ -117,6 +122,25 @@ func (c *secretUpsertCommand) Init(args []string) error {
 		return errors.NotValidf("secret owner %q", c.owner)
 	}
 
+	// Process keys where content is to come from a file.
+	const fromFile = "#file"
+	for i, arg := range args {
+		idx := strings.Index(arg, "=")
+		if idx <= 0 {
+			continue
+		}
+		possibleKey := arg[0:idx]
+		if !strings.HasSuffix(possibleKey, fromFile) {
+			continue
+		}
+		key := strings.TrimSuffix(possibleKey, fromFile)
+		content, err := os.ReadFile(arg[idx+1:])
+		if err != nil {
+			return errors.Annotatef(err, "reading content for secret key %q", key)
+		}
+		args[i] = fmt.Sprintf("%s=%s", key, content)
```

Comment:
do we want to limit the max size of the file?

Reply:
yeah, added size check for individual keys plus entire content